util/store/store.go

```diff
@@ -164,6 +168,36 @@ func SetUint64(store sdk.KVStore, key []byte, val uint64, errField string) error
 	return nil
 }
 
+// GetInt64 retrieves an int64 from a KVStore, or returns zero if no value is stored.
+// Uses gogoproto Int64Value for unmarshaling, and panics if a stored value fails to unmarshal.
+// Accepts an additional string which should describe the field being retrieved in custom error messages.
+// Allows negative values.
+func GetInt64(store sdk.KVStore, key []byte, errField string) int64 {
+	if bz := store.Get(key); len(bz) > 0 {
+		return Int64(bz, errField)
+	}
+	// No stored bytes at key: return zero
+	return 0
+}
```

Comment:
let's use generics. There is lot of repeated code.

Reply:
Generics did not work here on my initial attempt - there were too many varying methods between the different types, including marshaling and even how to compare them to their zero values. We can make this a github issue but it shouldn't be a priority since it doesn't add features.

Reply:
I can probably try smaller generic implementations (at least merge uint32 and uint64, for example) instead of trying to also merge sdk.Int and other number types too. My initial attempt was probably too ambitious

Reply:
I think all the native types can be done as generics. Same all types for which we call `.Unmarshal(..)`

Reply:
LMK, and if you want I can experiment with that.

Reply:
already exists as #1870 , so I'll resolved this comment